plugins/ldap/server/__init__.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import jsonschema
+import ldap
+import six
+
+from girder import events, logger
+from girder.api import access
+from girder.api.describe import autoDescribeRoute, Description
+from girder.api.rest import boundHandler
+from girder.models.model_base import ValidationException
+from girder.utility import setting_utilities
+from girder.utility.model_importer import ModelImporter
+from .constants import PluginSettings
+
+_LDAP_ATTRS = ('uid', 'mail', 'cn', 'sn', 'givenName', 'distinguishedName')
+_MAX_NAME_ATTEMPTS = 10
+_CONNECT_TIMEOUT = 4  # seconds
+_serversSchema = {
+    'type': 'array',
+    'items': {
+        'type': 'object',
+        'properties': {
+            'uri': {
+                'type': 'string',
+                'minLength': 1
+            },
+            'bindName': {
+                'type': 'string',
+                'minLength': 1
+            },
+            'baseDn': {
+                'type': 'string'
+            }
+        },
+        'required': ['uri', 'bindName', 'baseDn']
+    }
+}
+
+
+@setting_utilities.default(PluginSettings.LDAP_SERVERS)
+def _defaultServers():
+    return []
+
+
+@setting_utilities.validator(PluginSettings.LDAP_SERVERS)
+def _validateLdapServers(doc):
+    try:
+        jsonschema.validate(doc['value'], _serversSchema)
+    except jsonschema.ValidationError as e:
+        raise ValidationException('Invalid LDAP servers list: ' + e.message)
```

Comment:
In this case I believe bindName is being passed, but it's the empty string, and minLength is the rule used to enforce a non-empty string.